htlcswitch/link.go

```diff
@@ -1045,9 +1075,16 @@ func (l *channelLink) processLockedInHtlcs(
 						btcutil.Amount(pd.Amount-fwdInfo.AmountToForward),
 						btcutil.Amount(expectedFee))
 
-					// TODO(andrew.shvv): send proper back error
-					reason := []byte{byte(lnwire.IncorrectValue)}
-					l.sendHTLCError(pd.RHash, reason)
+					var failure lnwire.FailureMessage
+					update, err := l.createChannelUpdate()
+					if err != nil {
+						failure = lnwire.NewTemporaryChannelFailure(nil)
+					} else {
+						failure = lnwire.NewFeeInsufficient(expectedFee,
```

Comment:
The value here is meant to be the the value of the _incoming_ HTLC, not the fee we expect.